google-cloud-clients/google-cloud-firestore/src/main/java/com/google/cloud/firestore/FirestoreImpl.java

```diff
@@ -352,7 +352,24 @@ public void onSuccess(List<WriteResult> writeResults) {
                   @Override
                   public void run() {
                     try {
-                      callbackResult.set(transactionCallback.updateCallback(transaction));
+                      T callbackValue = transactionCallback.updateCallback(transaction);
+                      if (callbackValue instanceof ApiFuture) {
+                        ApiFutures.addCallback(
```

Comment:
`ApiFutures#addCallback` is deprecated. Please use the same method with 3 parameters:
`ApiFutures#addCallback(ApiFuture<V> future, final ApiFutureCallback<? super V> callback, Executor executor)`.